frontend/src/metabase/entities/fields.js

```diff
@@ -221,4 +241,32 @@ const Fields = createEntity({
   ),
 });
 
+const useGetFetchFieldValuesQuery = query => {
+  const tableId = query.table_id;
+  const result = useGetFieldValuesQuery(
+    query === skipToken ? skipToken : query.id,
+  );
+
+  const { data } = result;
+  const transformedData = useMemo(() => {
+    return data ? transformFieldValuesData(data, tableId) : data;
+  }, [data, tableId]);
+
+  return useMemo(
+    () => ({ ...result, data: transformedData }),
+    [result, transformedData],
+  );
```

Comment:
All that to mimic this: https://github.com/metabase/metabase/blob/f1607ca2138e944516ec4c9e4f91448c84715a06/frontend/src/metabase/entities/fields.js#L117-L125